### ISSUE TITLE
btf: `BTF_KIND_FLOAT` support is missing

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [cilium/ebpf#343]
https://github.com/torvalds/linux/commit/8fd886911a6a9 added support for a new `BTF_KIND_FLOAT` type. This library is missing this support and hence BTF related functionality is not working on Linux 5.13:

```
... load kernel spec: can't read types: type id 9468: unknown kind: Unknown
```

**Comment by lmb** [comment]
This will need changes to:
* `readTypes`: add kindFloat switch case
* `inflateRawTypes`: add kindFloat switch case
* Add `type Float struct` to package `btf`

**Comment by mauriciovasquezbernal** [comment]
I gave it a try in https://github.com/cilium/ebpf/pull/345 but I haven't been able to update `internal/btf/testdata/vmlinux-btf.gz` because I get so many errors when trying to compile that kernel using https://github.com/cilium/ci-kernels. 

I have the following diff:
```
$ git diff
diff --git make.sh make.sh
index c96f2b6..b705c44 100755
--- make.sh
+++ make.sh
@@ -30,7 +30,7 @@ fetch_and_configure() {
        echo "${src_dir}"
 }
 
-readonly kernel_versions=("4.9.266" "4.14.230" "4.19.187" "5.4.112" "5.10.30")
+readonly kernel_versions=("4.9.266" "4.14.230" "4.19.187" "5.4.112" "5.10.30" "5.13.1")
 for kernel_version in "${kernel_versions[@]}"; do
        series="$(echo "$kernel_version" | cut -d . -f 1-2)"
```

When I run `make.sh`, the compilation appears to work fine but then the following error happens:

```
...
WARNING: Symbol version dump "Module.symvers" is missing.
         Modules may not have dependencies or modversions.
         You may get many unresolved symbol warnings.
  MODPOST /home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/Module.symvers
WARNING: modpost: "__x86_indirect_alt_call_rax" [/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko] undefined!
WARNING: modpost: "trace_event_raw_init" [/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko] undefined!
WARNING: modpost: "trace_event_reg" [/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko] undefined!
WARNING: modpost: "sysfs_remove_bin_file" [/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko] undefined!
WARNING: modpost: "sysfs_create_bin_file" [/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko] undefined!
WARNING: modpost: "kernel_kobj" [/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko] undefined!
WARNING: modpost: "bpf_trace_run2" [/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko] undefined!
WARNING: modpost: "trace_handle_return" [/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko] undefined!
WARNING: modpost: "trace_event_printf" [/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko] undefined!
WARNING: modpost: "trace_raw_output_prep" [/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko] undefined!
WARNING: modpost: suppressed 12 unresolved symbol warnings because there were too many)
make[4]: *** No rule to make target 'scripts/module.lds', needed by '/home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.ko'.  Stop.
make[4]: *** Waiting for unfinished jobs....
  CC [M]  /home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod/bpf_testmod.mod.o
  BINARY   xdpxceiver
  CC      /home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/tools/build/bpftool/cfg.o
  CC      /home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/tools/build/bpftool/map.o
  GEN     /home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/tools/build/bpftool//libbpf/bpf_helper_defs.h
  BINARY   test_verifier
  MKDIR   /home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/tools/build/bpftool//libbpf/staticobjs/
  CC      /home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/tools/build/bpftool//libbpf/staticobjs/libbpf.o
  BINARY   test_tag
make[3]: *** [scripts/Makefile.modpost:156: __modpost] Error 2
make[2]: *** [Makefile:1762: modules] Error 2
make[1]: *** [Makefile:16: all] Error 2
make: *** [Makefile:174: /home/mvb/kinvolk/ebpf/ci-kernels/build/linux-5.13.1/tools/testing/selftests/bpf/bpf_testmod.ko] Error 2
make: *** Waiting for unfinished jobs....
```

I have no idea how to solve it...

**Comment by lmb** [comment]
The fix for that problem is in https://github.com/cilium/ci-kernels/pull/13

However, compiling 5.13 selftests is still broken: https://lore.kernel.org/bpf/CACAyw99ZrBRr9QydPVvuWNksGfOckq-giTUR29sjzZDfdx5MFA@mail.gmail.com/

**Comment by lmb** [comment]
Fixed with f7c00a7b0e39904bf62aa83910b55dd6c82fd834
